.github/workflows/order-id-validator.yml

```diff
@@ -0,0 +1,24 @@
+name: Order ID validation
+
+on:
+  issues:
+    types: [opened]
+
+jobs:
+  validate-order-id:
+    runs-on: ubuntu-latest
+    name: Label issue based on order ID
+    steps:
+      - uses: actions/checkout@v3
+      - name: Label issue
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const script = require('./scripts/orderIdValidation.js')
+            await script(core, github)
+        env:
+          OWNER: ${{ github.repository_owner }}
+          REPO: ${{ github.event.repository.name }}
+          ISSUE_NUMBER: ${{ github.event.issue.number }}
+          TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ORDER_API_TOKEN: ${{ secrets.SHOP_AUTH_TOKEN }}
```

Comment:
@hasdfa we have the following keys in use today

<img src="https://github.com/user-attachments/assets/f0bbb684-4fb5-4e13-9b44-38b4391e4c1a" width="450" />

https://store-wp.mui.com/wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys

It feels like we should create a new one specifically for GitHub Actions so we can reduce disruption in case of a security leak. How about we make the change?

@michelengelen which key did you use?

Reply:
@oliviertassinari I did use the one from the private toolpad app

Reply:
Ok, I have created an organization secret for the GitHub action, easier to rorate this way.

Reply:
I broke the GitHub Action with this token change. We should be back up and running with 113a61207d8afa836bce60a3a2c002d38cec0e9c. It now shows a clear error in case our token gets revoked in the future: 

<img width="766" alt="SCR-20240828-upvz" src="https://github.com/user-attachments/assets/8da219d5-d08d-49ed-97cd-2d3742f29c6f">

Source: https://github.com/oliviertassinari/mui-x/actions/runs/10604955513/job/29392734749

It also now doesn't assume that `Basic ` is in the token. This is normally not how it's done https://github.com/search?q=org%3Amui+%22Authorization%3A+%22&type=code. The only exception that I can find are this one (gone) and one done by Prakhar: https://github.com/mui/mui-private/blob/04c21bf04d29f9985ea400ef52afb3861203867e/tools-private/toolpad/resources/functions.ts#L107.